drivers/sdp3x/sdp3x.c

```diff
@@ -0,0 +1,458 @@
+    }
+    xtimer_usleep(2000);
+    ret = i2c_write_bytes(DEV_I2C, DEV_ADDR, ptr, 0, 0);
+    i2c_release(DEV_I2C);
+
+    return ret;
+}
+
+/**
+ *      intern function to read temperature
+ *      @param  dev      device
+ *      @return Temperature value in celsius
+ */
+int32_t _SDP3x_read_temp(const sdp3x_t *dev)
+{
+    int16_t data[3];
+    uint8_t ret = _SDP3x_read_data(dev, data);
+
+    if (ret != 0) {
+        DEBUG("[SDP3x] Error reading date\n");
+        return ret;
+    }
+    DEBUG("[SDP3x] read temperature bare value %i\n", data[1]);
+    return _SDP3x_convert_to_celsius(data[1]);
+}
+
+/**
+ *      intern function to read differential pressure
+ *      @param  dev      device
+ *      @return Pressure value in pascals
+ */
+int32_t _SDP3x_read_pressure(const sdp3x_t *dev)
+{
+    int16_t data[3];
+    uint8_t ret = _SDP3x_read_data(dev, data);
+
+    if (ret != 0) {
+        DEBUG("[SDP3x] Error reading date\n");
+        return ret;
+    }
+    return _SDP3x_convert_to_pascal(dev, data[0]);
+}
+
+/** Read measurements
+ *
+ *      Data consists of:
+ *      2 byte Differential Pressure,
+ *      1 byte CRC
+ *      2 byte Temperature,
+ *      1 byte CRC
+ *      2 byte Scale Factor differential pressure
+ *      1 byte CRC
+ *
+ *      @param addr   Address byte 1 = 0x21, 2 = 0x22, 3 = 0x23,
+ *                    else = Last address
+ *      @param data   Data will be stored here
+ *
+ *      @return       0 if data could be read, 1 if CRC-Error
+ */
+static int8_t _SDP3x_read_data(const sdp3x_t *dev, int16_t *data)
+{
+    int ret = 0;
+    uint8_t readData[9];
+
+    i2c_acquire(DEV_I2C);
+    ret = i2c_read_bytes(DEV_I2C, DEV_ADDR, readData, sizeof(readData), 0);
+    i2c_release(DEV_I2C);
+
+    if (ret != 0) {
+        DEBUG("[SDP3x] read_data: ret %i\n", ret);
+        return ret;
+    }
+    data[2] = (readData[6] << 8) + readData[7];
+    data[1] = (readData[3] << 8) + readData[4];
+    data[0] = (readData[0] << 8) + readData[1];
+
+    uint8_t correct = 1;
+
+    correct &= _checkCRC(data[2], readData[8]);
+    correct &= _checkCRC(data[1], readData[5]);
+    correct &= _checkCRC(data[0], readData[2]);
+
+    if (correct == 0) {
+        DEBUG("[SDP3x] read_data: CRC-Failure\n");
+        return SDP3x_CRCERROR;
+    }
+
+    return 0;
+}
+
+/**
+ *      intern method to convert sensor value to pascal
+ *      @param  dev      device
+ *      @param  value    raw sensor value
+ *      @return pressure value in centipascal
+ */
+static int32_t _SDP3x_convert_to_pascal(const sdp3x_t *dev, int16_t value)
+{
+    int16_t div = 1;
+
+    if (DEV_MODEL == 1) {
+        div = 60;
+    }
+    else {
+        div = 240;
+    }
+    return ((value * 100) / div);
+}
+
+/**
+ *      intern method to convert sensor value to celsius
+ *      @param  dev     device
+ *      @param  value   raw sensor value
+ *      @return celsius value * 100
+ */
+static int32_t _SDP3x_convert_to_celsius(int16_t value)
+{
+    int16_t div = 200;
```

Comment:
so div should be `200 * dp_scale_factor` - or is `dp_scale_factor` assumed to be 200 here?